index.d.ts

```diff
@@ -0,0 +1,195 @@
+	 */
+	readonly version?: string | false;
+
+	/**
+	 * Automatically show the help text when the `--help` flag is present. Useful to set this value to `false` when a CLI manages child CLIs with their own help text.
+	 */
+	readonly autoHelp?: boolean;
+
+	/**
+	 * Automatically show the version text when the `--version` flag is present. Useful to set this value to `false` when a CLI manages child CLIs with their own version text.
+	 */
+	readonly autoVersion?: boolean;
+
+	/**
+	 * package.json as an `Object`. Default: Closest package.json upwards
+	 *
+	 * *You most likely don't need this option.*
+	 */
+	readonly pkg?: {[key: string]: unknown};
+
+	/**
+	 * Custom arguments object.
+	 *
+	 * @default process.argv.slice(2)
+	 */
+	readonly argv?: ReadonlyArray<string>;
+
+	/**
+	 * Infer the argument type.
+	 *
+	 * By default, the argument `5` in `$ foo 5` becomes a string. Enabling this would infer it as a number.
+	 *
+	 * @default false
+	 */
+	readonly inferType?: boolean;
+
+	/**
+	 * Value of `boolean` flags not defined in `argv`. If set to `undefined` the flags not defined in `argv` will be excluded from the result. The `default` value set in `boolean` flags take precedence over `booleanDefault`.
+	 *
+	 * **Caution: Explicitly specifying undefined for `booleanDefault` has different meaning from omitting key itself.**
+	 *
+	 * @example
+	 *
+	 * const cli = meow(`
+	 *	Usage
+	 *	  $ foo
+	 *
+	 *	Options
+	 *	  --rainbow, -r  Include a rainbow
+	 *	  --unicorn, -u  Include a unicorn
+	 *	  --no-sparkles  Exclude sparkles
+	 *
+	 *	Examples
+	 *	  $ foo
+	 *	  🌈 unicorns✨🌈
+	 * `, {
+	 *	booleanDefault: undefined,
+	 *	flags: {
+	 *		rainbow: {
+	 *			type: 'boolean',
+	 *			default: true,
+	 *			alias: 'r'
+	 *		},
+	 *		 unicorn: {
+	 *			type: 'boolean',
+	 *			default: false,
+	 *			alias: 'u'
+	 *		},
+	 *		cake: {
+	 *			type: 'boolean',
+	 *			alias: 'c'
+	 *		},
+	 *		sparkles: {
+	 *			type: 'boolean',
+	 *			default: true
+	 *		}
+	 *	}
+	 * });
+	 *
+	 * //{
+	 * //	flags: {
+	 * //		rainbow: true,
+	 * //		unicorn: false,
+	 * //		sparkles: true
+	 * //	},
+	 * //	unnormalizedFlags: {
+	 * //		rainbow: true,
+	 * //		r: true,
+	 * //		unicorn: false,
+	 * //		u: false,
+	 * //		sparkles: true
+	 * //	},
+	 * //	…
+	 * //}
+	 */
+	readonly booleanDefault?: boolean | null;
+
+	/**
+	 * Whether to use [hard-rejection](https://github.com/sindresorhus/hard-rejection) or not. Disabling this can be useful if you need to handle `process.on('unhandledRejection')` yourself.
+	 *
+	 * @default true
+	 */
+	readonly hardRejection?: boolean;
+}
+
+export interface Result {
+	/**
+	 * Non-flag arguments.
+	 */
+	input: string[];
+
+	/**
+	 * Flags converted to camelCase excluding aliases.
+	 */
+	flags: {[name: string]: unknown};
+
+	/**
+	 * Flags converted camelCase including aliases.
+	 */
+	unnormalizedFlags: {[name: string]: unknown};
+
+	/**
+	 * The `package.json` object.
+	 */
+	pkg: {[key: string]: unknown};
```

Comment:
Would be really cool if we could properly type the package.json.

https://docs.npmjs.com/files/package.json
http://json.schemastore.org/package

You could submit the definition to https://github.com/sindresorhus/type-fest so we could reuse it in other packages too.

What do you think?

---

We could also expose a type that extends the spec compliant one with a realistic definition that also supports keys like `types` and `eslintConfig`.

Reply:
Sounds reasonable. Will do.